src/Ragel.jl

```diff
@@ -229,18 +229,18 @@ end
 # Open functions for various sources
 # ----------------------------------
 
-function Base.open{T<:FileFormat}(filename::AbstractString, ::Type{T}; args...)
+function Base.open{T<:FileFormat}(filename::AbstractString, ::Type{T}, args...; kwargs...)
     memory_map = false
     i = 0
-    for arg in args
+    for arg in kwargs
         i += 1
         if arg[1] == :memory_map
```

Comment:
This does not mean `memory_map` must be the first argument. `arg[1]` is a key of a keyword argument and `arg[2]` will be a value of it. This loop tries to find a pair of `memory_map => true/false` and thus `memory_map` can be located at any position.